File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string temp = "";
        for(char c : s){
            if(isalnum(c)){
                temp += tolower(c);
            }
        }
        int i = 0;
        int j = temp.size() - 1;
        while(i < j){
            if(temp[i] != temp[j]){
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
};

// class Solution {
// public:
//     bool isPalindrome(string s) {
//         string temp = "";
//         for(char c : s){
//             if(isalnum(c)){
//                 temp += tolower(c);
//             }
//         }
//         int left = 0;
//         int right = temp.size() - 1;
//         while(left < right){
//             if(temp[left] != temp[right]){
//                 return false;
//             }
//             left++;
//             right--;
//         }
//         return true;
//     }
// };

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.toLowerCase();
        String rev = "";
    for(int i = 0; i < s.length(); i++){
        if(Character.isLetterOrDigit(s.charAt(i))){
            rev += s.charAt(i);
        }
    }
    int left = 0;
    int right = rev.length() - 1;
    while(left < right){
        if(rev.charAt(left)!=rev.charAt(right)){
            return false;
        }
        left++;
        right--;
    }
    return true;
    }
}